### PR TITLE
RS migration - trymerge to upload merge records to s3

### PR DESCRIPTION
**Pull request by clee2000**
Uploads merge records to to ossci-raw-job-status (public) bucket instead of directly to rockset

The runner used by trymerge is a GH runner, so it doesn't have access to s3.  Instead, I save the record as a json and upload the json to s3 in a different step that runs after the aws credentials are configured.

The role is defined [here](https://togithub.com/pytorch-labs/pytorch-gha-infra/pull/421)

### CODE REVIEW

File: .github/scripts/trymerge.py [hunk 1]
```diff
@@ -1505,59 +1504,41 @@ def save_merge_record(
...
-                "merge_commit_sha": merge_commit_sha,
-                "is_failed": is_failed,
-                "skip_mandatory_checks": skip_mandatory_checks,
-                "ignore_current": ignore_current,
-                "error": error,
-            }
-        ]
 
-        client = rockset.RocksetClient(
-            host="api.usw2a1.rockset.com", api_key=os.environ["ROCKSET_API_KEY"]
```
Comment (huydhn) [inline .github/scripts/trymerge.py:1550]: Note: Remind me to delete the key after this lands

**Review by huydhn — APPROVED** [review]

File: .github/workflows/trymerge.yml [hunk 2]
```diff
@@ -84,6 +84,22 @@ jobs:
           set -x
           python3 .github/scripts/comment_on_pr.py "${PR_NUM}" "merge"
 
+      - name: configure aws credentials
+        uses: aws-actions/configure-aws-credentials@v3
+        continue-on-error: true
+        with:
+          role-to-assume: arn:aws:iam::749337293305:role/upload_to_ossci_raw_job_status
```
Comment (huydhn) [inline .github/workflows/trymerge.yml:91]: This looks like a wrong AWS account ID.  Should is be `308535385114` to match what is in https://github.com/pytorch-labs/pytorch-gha-infra/pull/421?

Comment (ZainRizvi) [reply .github/workflows/trymerge.yml:91]: 308535385114 is the ossci AWS account.  749337293305 is the pytorch account.  

The bucket lives in the ossci account

**Review by ZainRizvi — APPROVED** [review]
For now do we have a way to ingest this data back into Rockset, or is the collection not read by any lambda query??

**Review by malfet — APPROVED** [review]

**Pull request closed without merging**